src/bfx_tools/pyensembl.ml

```diff
@@ -0,0 +1,29 @@
+open Biokepi_run_environment
+open Common
+
+let cache_genome ~(run_with: Machine.t) ~reference_build =
+  let open KEDSL in
+  let pyensembl =
+    Machine.get_tool run_with Machine.Tool.Definition.(create "pyensembl")
+  in
+  let genome = Machine.(get_reference_genome run_with reference_build) in
+  let ensembl_release = genome |> Reference_genome.ensembl in
+  let species = genome |> Reference_genome.species in
+  let name = sprintf "pyensembl_cache-%d-%s" ensembl_release species in
+  workflow_node
+    without_product
+    ~name
+    ~edges:[
+      depends_on Machine.Tool.(ensure pyensembl);
+    ]
+    ~make:(
+      Machine.run_download_program run_with
+        ~requirements:[`Internet_access;]
+        ~name
+        Program.(
+          Machine.Tool.(init pyensembl)
+          && shf "pyensembl install --release %d --species \"%s\""
+             ensembl_release
+             species
```

Comment:
`pyensembl` has a cache directory for each genome. Would you want to pass a custom directory instead?